src/tokenizer.rs

```diff
@@ -1147,7 +1147,11 @@ impl<'a> Tokenizer<'a> {
                         s.push('.');
                         chars.next();
                     }
-                    s += &peeking_take_while(chars, |ch| ch.is_ascii_digit());
+
+                    s += &peeking_take_while(chars, |ch| {
+                        ch.is_ascii_digit()
+                            || self.dialect.supports_numeric_literal_underscores() && ch == '_'
```

Comment:
What happens if starts or ends with `_`? 

I think Clickhouse rejects that: https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/StringUtils.h#L171-L182

Reply:
It could be worth having a test to demonstrate the behavior with a trailing underscore